jumanji/environments/logic/sudoku/viewer.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence, Tuple
+
+import matplotlib
+import matplotlib.pyplot as plt
+
+import jumanji
+from jumanji.environments.logic.sudoku.constants import BOARD_WIDTH
+from jumanji.environments.logic.sudoku.env import State
+from jumanji.viewer import Viewer
+
+
+class SudokuViewer(Viewer[State]):
+    def __init__(
+        self,
+        name: str = "Sudoku",
+    ) -> None:
+        self._name = name
+        self._animation: Optional[matplotlib.animation.Animation] = None
+
+    def render(
+        self,
+        state: State,
+        save_path: Optional[str] = None,
+        ax: Optional[plt.Axes] = None,
+    ) -> None:
+        self._clear_display()
+
+        if ax is None:
+            fig, ax = plt.subplots(figsize=(6, 6))
+            plt.title(f"{self._name}")
+        else:
+            fig = ax.figure
+            ax.clear()
+        self._draw(ax, state)
+        # Uncomment to display the remaining-actions number.
```

Comment:
Rather than doing this in a comment wouldn't it be better to have a configurable arg (off by default say)?

Reply:
Deleted